src/compute-client/src/controller/instance.rs

```diff
@@ -887,13 +887,19 @@ where
         &mut self,
         dataflow: DataflowDescription<mz_compute_types::plan::Plan<T>, (), T>,
     ) -> Result<(), DataflowCreationError> {
-        // Validate the dataflow as having inputs whose `since` is less or equal to the dataflow's `as_of`.
-        // Start tracking frontiers for each dataflow, using its `as_of` for each index and sink.
         let as_of = dataflow
             .as_of
             .as_ref()
             .ok_or(DataflowCreationError::MissingAsOf)?;
 
+        // If the `as_of` is empty, then don't even create a dataflow.
+        if as_of.is_empty() {
+            return Ok(());
+        }
```

Comment:
Oh, you still have to create the `CollectionState` and do all the setup work!

I think what we want to do when the `as_of` is empty is:
* Set all the `replica_write_frontier`s to the empty frontier too. That should ensure that we install no read holds on the dependencies (but verify!).
* Don't send the `CreateDataflow` command.

And I think we also need to return an error when any of the dataflow exports is something else than a `persist_sink`. Maybe we can allow indexes too, given that you cannot query an index with an empty read frontier. But subscribes and copy-tos we have to reject, I think. The client expects responses for those that we won't be able to deliver.

Reply:
Yes, on it!